src/main/scala/educational/Monad.scala

```diff
@@ -15,16 +15,20 @@ import functor.FunctorSimpleImpl.Functor
 
 2) unit must be identity for join
 
-         unit
-     A ---------> M[A]
-     |             |
-unit |             |  unit
-     |             |
-     \/            \/
- M[A] <------  M[M[A]]
-         join
+            pure
+        A ---------> M[A]
+        |             |
+flatten |             |  pure
+        |             |
+        \/            \/
```

Comment:
should not be the arrow reverted to point from M[A] to A?

Reply:
Thank you @plokhotnyuk for spotting the problem!

We can collapse 2 `layers of monad` into one - go from F[F[A]] into F[A]. We cannot extract value from monad - you know we need a trap ;) https://twitter.com/impurepics?lang=en So I cannot move from F[A] to A using flatten.

But I can go from A to F[A] using pure.